pkg/config/config.go

```diff
@@ -175,29 +243,136 @@ func (lci *LocalConfigInfo) DeleteConfiguration(parameter string) error {
 
 		switch parameter {
 		case "cpu":
-			lci.ComponentSettings.MinCPU = nil
-			lci.ComponentSettings.MaxCPU = nil
+			lci.componentSettings.MinCPU = nil
+			lci.componentSettings.MaxCPU = nil
 		case "memory":
-			lci.ComponentSettings.MinMemory = nil
-			lci.ComponentSettings.MaxMemory = nil
+			lci.componentSettings.MinMemory = nil
+			lci.componentSettings.MaxMemory = nil
 		default:
-			if err := util.DeleteConfiguration(&lci.ComponentSettings, parameter); err != nil {
+			if err := util.DeleteConfiguration(&lci.componentSettings, parameter); err != nil {
 				return err
 			}
 		}
-		return util.WriteToFile(&lci.LocalConfig, lci.Filename)
+		return writeToFile(&lci.LocalConfig, lci.Filename)
 	}
 	return errors.Errorf("unknown parameter :'%s' is not a parameter in local odo config", parameter)
 
 }
 
+// GetComponentSettings returns the componentSettings from local config
+func (lci *LocalConfigInfo) GetComponentSettings() ComponentSettings {
+	return lci.componentSettings
+}
+
+// SetComponentSettings sets the componentSettings from to the local config and writes to the file
+func (lci *LocalConfigInfo) SetComponentSettings(cs ComponentSettings) error {
+	lci.componentSettings = cs
+	return writeToFile(&lci.LocalConfig, lci.Filename)
+}
+
 // GetComponentType returns type of component (builder image name) in the config
 // and if absent then returns default
 func (lc *LocalConfig) GetComponentType() string {
-	if lc.ComponentSettings.ComponentType == nil {
+	if lc.componentSettings.ComponentType == nil {
+		return ""
+	}
+	return *lc.componentSettings.ComponentType
+}
+
+// GetPath returns the path, returns default if nil
+func (lc *LocalConfig) GetPath() string {
+	if lc.componentSettings.Path == nil {
+		return ""
+	}
+	return *lc.componentSettings.Path
+}
+
+// GetRef returns the ref, returns default if nil
+func (lc *LocalConfig) GetRef() string {
+	if lc.componentSettings.Ref == nil {
+		return ""
+	}
+	return *lc.componentSettings.Ref
+}
+
+// GetSourceType returns the source type, returns default if nil
+func (lc *LocalConfig) GetSourceType() string {
```

Comment:
Can this return CreateType or SourceType or what ever it is please?

Reply:
you are looking at outdated code @anmolbabu